File: practica2/push.cc
#include "push.h"
#include <iostream>


using namespace std;

    push::push()    
    {    
    }  

    void ejecutar(stack<int> &pila) // el pedir un argumento detras de la función no esta implementado no se hacerlo, ver como se hace
    {
        int argumento;
        cin >> argumento;

        pila.push(argumento);
    }

File: practica2/java/Push.java
import java.util.Stack;

/**
 * push
 */
public class Push extends Inst{

    private int argumento;

    public Push(int param) {argumento = param;}

	@Override
    public int ejecutar(Stack<Integer> pila, int pc){
        pila.push(argumento);
        pc++;
        return pc;
    };

	@Override
    public String nombre(){
        return "push " + Integer.toString(argumento);
    }
}
